src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -176,6 +186,11 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 if (valueProvider != null)
                 {
                     newBindingContext.ValueProvider = valueProvider.Filter(bindingSource);
+                    if (newBindingContext.ValueProvider == null)
+                    {
+                        // Unable to find a value provider for this binding source.
+                        return null;
+                    }
                 }
```

Comment:
If / when we fix #2900, can move this block to the top and avoid creating the `newBindingContext` right above where we throw it away and return `null`. Reordering things today visibly changes the behaviour, hiding some expected errors.